src/hooks/useUserOnboardingContext.ts

```diff
@@ -15,54 +15,107 @@ limitations under the License.
 */
 
 import { logger } from "matrix-js-sdk/src/logger";
-import { ClientEvent, IMyDevice, Room } from "matrix-js-sdk/src/matrix";
-import { useCallback, useEffect, useState } from "react";
+import { ClientEvent } from "matrix-js-sdk/src/matrix";
+import { useCallback, useEffect, useRef, useState } from "react";
 
 import { MatrixClientPeg } from "../MatrixClientPeg";
+import { Notifier } from "../Notifier";
 import DMRoomMap from "../utils/DMRoomMap";
-import { useEventEmitter } from "./useEventEmitter";
 
 export interface UserOnboardingContext {
-    avatar: string | null;
-    myDevice: string;
-    devices: IMyDevice[];
-    dmRooms: {[userId: string]: Room};
+    hasAvatar: boolean;
+    hasDevices: boolean;
+    hasDmRooms: boolean;
+    hasNotificationsEnabled: boolean;
+}
+
+const USER_ONBOARDING_CONTEXT_INTERVAL = 5000;
+
+/**
+ * Returns a persistent, non-changing reference to a function
+ * This function proxies all its calls to the current value of the given input callback
+ *
+ * This allows you to use the current value of e.g., a state in a callback that’s used by e.g., a useEventEmitter or
+ * similar hook without re-registering the hook when the state changes
+ * @param value changing callback
+ */
+function useRefOf<T extends [], R>(value: (...values: T) => R): (...values: T) => R {
+    const ref = useRef(value);
+    ref.current = value;
+    return useCallback(
+        (...values: T) => ref.current(...values),
+        [],
+    );
 }
 
 export function useUserOnboardingContext(): UserOnboardingContext | null {
     const [context, setContext] = useState<UserOnboardingContext | null>(null);
 
     const cli = MatrixClientPeg.get();
-    const handler = useCallback(async () => {
-        try {
-            const profile = await cli.getProfileInfo(cli.getUserId());
-
-            const myDevice = cli.getDeviceId();
-            const devices = await cli.getDevices();
-
-            const dmRooms = DMRoomMap.shared().getUniqueRoomsWithIndividuals() ?? {};
-            setContext({
-                avatar: profile?.avatar_url ?? null,
-                myDevice,
-                devices: devices.devices,
-                dmRooms: dmRooms,
-            });
-        } catch (e) {
-            logger.warn("Could not load context for user onboarding task list: ", e);
-            setContext(null);
-        }
-    }, [cli]);
-
-    useEventEmitter(cli, ClientEvent.AccountData, handler);
+    const handler = useRefOf(
+        useCallback(async () => {
```

Comment:
Seems weird to use both, the point of `useCallback` is to provide a semi-stable reference function, `useRefOf` provides a fully-stable reference function